runtime/lua/vim/lsp.lua

```diff
@@ -1255,26 +1299,21 @@ end
 ---        returns `(nil, err)` where `err` is a string describing the failure
 ---        reason.
 function lsp.buf_request_sync(bufnr, method, params, timeout_ms)
-  local request_results = {}
-  local result_count = 0
-  local function _sync_handler(err, _, result, client_id)
-    request_results[client_id] = { error = err, result = result }
-    result_count = result_count + 1
-  end
-  local client_request_ids, cancel = lsp.buf_request(bufnr, method, params, _sync_handler)
-  local expected_result_count = 0
-  for _ in pairs(client_request_ids) do
-    expected_result_count = expected_result_count + 1
-  end
+  local request_results
+
+  local cancel = lsp.buf_request_all(bufnr, method, params, function(it)
+    request_results = it
+  end)
 
   local wait_result, reason = vim.wait(timeout_ms or 100, function()
-    return result_count >= expected_result_count
-  end, 10)
+    return request_results ~= nil
+  end)
```

Comment:
I believe the implicit default for this will now be 200 ms (over the default 100 ms timeout window) is that what you wanted?

Reply:
Thats a mistake